### ISSUE TITLE
Zed inserts surrounding spaces to JS/TS objects while typing.

### ISSUE THREAD

**Issue by nathanael-ruf** [zed-industries/zed#15405]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

https://github.com/user-attachments/assets/3c60fa4a-e686-4416-bd1e-ab569d8ac599

I expect no automatic formatting while typing.
I tried disabling prettier in settings with no success.
Who/what is causing this? Is there a way to turn if off or is this a bug?

### Environment

Zed: v0.146.1 (Zed Preview)
OS: macOS 14.3.1
Memory: 16 GiB
Architecture: aarch64

**Comment by SomeoneToIgnore** [comment]
Could be 

```json
// Whether to use additional LSP queries to format (and amend) the code after
// every "trigger" symbol input, defined by LSP server capabilities.
"use_on_type_format": true,
```

**Comment by nathanael-ruf** [comment]
Thanks, that seems to be it. However, it only works if I put `"use_on_type_format": false` in the global settings.
Putting it under
```json
"languages": {
    "TypeScript": {
        "use_on_type_format": false
    }
} 
```
has no effect (that's what I tried before opening this issue). Is that supposed to work?

**Comment by SomeoneToIgnore** [comment]
Good catch, it's on the editor (global) settings level now, for some reason: https://github.com/zed-industries/zed/blob/78a2539d5961ecfcf829aa67270cbdc56df7c0e1/crates/editor/src/editor.rs#L3260

Makes sense to make it language-levelled indeed.
